### PR TITLE
Feature: add sync meta yml gitbook

### LINKED ISSUE #20: Push meta.yml to Gitbook

### PR DESCRIPTION
**Pull request by huyu335**

### CODE REVIEW

**Comment by huyu335** [comment]
Example: https://app.gitbook.com/o/Txb2lda7D1fX9CVuQbQ0/s/teTZ78hswvAaDZGDOBhN/polysolver
Turns out gitbook does not support yaml format unless we put it in a code block.

**Comment by rhshah** [comment]
Thank you for working on this Yu, can please clarify the structure of how the md would be stored in the docs branch? Sorry I am not following that via the github action. Would it be in the 'modules' directory and 'subworkflows' directory on top level

**Comment by huyu335** [comment]
> Thank you for working on this Yu, can please clarify the structure of how the md would be stored in the docs branch? Sorry I am not following that via the github action. Would it be in the 'modules' directory and 'subworkflows' directory on top level

Hi Ronak, yes, individual markdowns are here: https://github.com/mskcc-omics-workflows/modules/tree/docs/modules
Summary file: https://github.com/mskcc-omics-workflows/modules/blob/docs/SUMMARY.md
So in gitbook, all the modules and subworkflows are under the same page, as we discussed before.

**Comment by rhshah** [comment]
Did something go wrong with GBCMS? As that is not generated?

**Comment by huyu335** [comment]
> Did something go wrong with GBCMS? As that is not generated?

Yes, it failed in validation. We probably need to make some changes

**Comment by rhshah** [comment]
> > Did something go wrong with GBCMS? As that is not generated?
> 
> 
> 
> Yes, it failed in validation. We probably need to make some changes

I fixed yaml to markdown, specifically for the conversion part where we had certain things required which were not in the schema.can you regenerate the markdowns? Thank you again. The release is the same 0.0.1

**Comment by huyu335** [comment]
> > > Did something go wrong with GBCMS? As that is not generated?
> > 
> > 
> > Yes, it failed in validation. We probably need to make some changes
> 
> I fixed yaml to markdown, specifically for the conversion part where we had certain things required which were not in the schema.can you regenerate the markdowns? Thank you again. The release is the same 0.0.1

Thanks @rhshah ! I regenerated MDs for all modules in develop branch. However, subworkflow yaml is not able to be converted, example: https://github.com/mskcc-omics-workflows/modules/blob/develop/subworkflows/msk/bwa_markdup_bqsr/meta.yml
We probably need to use a different schema for it. Let me know if we need to include it in this PR, or do it later

**Comment by rhshah** [comment]
> > > > Did something go wrong with GBCMS? As that is not generated?
> > > 
> > > 
> > > Yes, it failed in validation. We probably need to make some changes
> > 
> > 
> > I fixed yaml to markdown, specifically for the conversion part where we had certain things required which were not in the schema.can you regenerate the markdowns? Thank you again. The release is the same 0.0.1
> 
> Thanks @rhshah ! I regenerated MDs for all modules in develop branch. However, subworkflow yaml is not able to be converted, example: https://github.com/mskcc-omics-workflows/modules/blob/develop/subworkflows/msk/bwa_markdup_bqsr/meta.yml We probably need to use a different schema for it. Let me know if we need to include it in this PR, or do it later

Hi Yu, Thank you, I have updated the scripts to handles subworkflow. You need to use the command as `python yaml_to_md.py convert -y test.yaml -S` for them. Validation remains the same the schema changes. The schema now is in respective folder for modules and subworkflows 
Screenshot 2024-02-02 at 3 13 27 PM

**Comment by rhshah** [comment]
@huyu335  also you might want to set the SCHEMA URL explicitly

File: .github/workflows/gitbook-sync.yml [hunk 1]
```diff
@@ -142,12 +142,12 @@ jobs:
         if: ${{ github.event_name == 'pull_request' }}
         run: |
           echo '${{ steps.cat_file_subworkflow.outputs.content }}' > ${{ github.workspace }}/temp.yml
-          python ${{ github.workspace }}/yaml_to_md.py validate --yaml-file ${{ github.workspace }}/temp.yml
+          python ${{ github.workspace }}/yaml_to_md.py validate --yaml-file ${{ github.workspace }}/temp.yml --schema-url https://raw.githubusercontent.com/mskcc-omics-workflows/yaml_to_md/0.0.1/nextflow_schema/subworkflows/meta-schema.json --subworkflows
```
Comment (rhshah) [inline .github/workflows/gitbook-sync.yml]: For validation we dont have that option, as its just another JSON to validate against

**Comment by huyu335** [comment]
> @huyu335 also you might want to set the SCHEMA URL explicitly

Thanks @rhshah !
MD for subworkflow is now generated successfully: https://app.gitbook.com/o/Txb2lda7D1fX9CVuQbQ0/s/x32s1acwH78k3hruISMe/subworkflows/bwa_markdup_bqsr
One small thing on the convertor side: https://github.com/mskcc-omics-workflows/yaml_to_md/blob/main/yaml_to_md.py#L26
This URL is no longer available, but it won't affect anything. For modules, I added in the schema path as well

**Comment by rhshah** [comment]
> > @huyu335 also you might want to set the SCHEMA URL explicitly
> 
> Thanks @rhshah ! MD for subworkflow is now generated successfully: https://app.gitbook.com/o/Txb2lda7D1fX9CVuQbQ0/s/x32s1acwH78k3hruISMe/subworkflows/bwa_markdup_bqsr One small thing on the convertor side: https://github.com/mskcc-omics-workflows/yaml_to_md/blob/main/yaml_to_md.py#L26 This URL is no longer available, but it won't affect anything. For modules, I added in the schema path as well

Ahh yes sorry, I am not sure if I want to make a whole release for that. I think its ok. What do you think?

File: .github/workflows/gitbook-sync.yml (hunk 1 above)
Comment (rhshah) [reply .github/workflows/gitbook-sync.yml]: @huyu335  i think once you fix this we can merge this

**Review by rhshah — APPROVED** [review]

**Review by rhshah — APPROVED** [review]

**Comment by huyu335** [comment]
> > > @huyu335 also you might want to set the SCHEMA URL explicitly
> > 
> > 
> > Thanks @rhshah ! MD for subworkflow is now generated successfully: https://app.gitbook.com/o/Txb2lda7D1fX9CVuQbQ0/s/x32s1acwH78k3hruISMe/subworkflows/bwa_markdup_bqsr One small thing on the convertor side: https://github.com/mskcc-omics-workflows/yaml_to_md/blob/main/yaml_to_md.py#L26 This URL is no longer available, but it won't affect anything. For modules, I added in the schema path as well
> 
> Ahh yes sorry, I am not sure if I want to make a whole release for that. I think its ok. What do you think?

yeah I agree. Let's not make change now

**Pull request merged**